### ISSUE TITLE
Better dot (DNS over TLS) support for resolvectl query

### ISSUE THREAD

**Issue by antipatico** [systemd/systemd#12859]
**Is your feature request related to a problem? Please describe.**
I would like to debug if I'm actually using DNSOverTLS or not. Moreover I would like being able to force DOT in `resolvectl query`.

**Describe the solution you'd like**
1. When resolving using `resolvectl query domain --legend true` and **cleartext DNS** the legend output should be:
```
domain: 1.2.3.4        -- link: eth0

-- Information acquired via protocol DNS in 69ms.
-- Data is authenticated: yes.
```
With the same command (and result), but using **DNS over TLS** the legend output should be:
```
domain: 1.2.3.4        -- link: eth0

-- Information acquired via protocol DOT in 69ms.
-- Data is authenticated: yes.
```

2. The `resolvectl query`'s `-p` flag should be able to support `dot` as a supported protocol, and enforce DNSOverTLS when used.

**Describe alternatives you've considered**
I haven't actually looked through neither the [resolved API](https://www.freedesktop.org/wiki/Software/systemd/resolved/) nor the resolvectl source itself, but granted resolvectl uses those API's, some extension of those *may* be needed.
The second feature seems apparently easier to implement since the merge of #12815, but again it is really API dependent.

**Comment by shibumi** [comment]
Small addition from me: DOT is not the protocol name. It's called "DNS over TLS", I think we should write "DNS over TLS" if possible.

**Comment by shibumi** [comment]
@antipatico some addition. It looks like systemd-resolved doesn't support cloudflare DNS servers. I don't know the reason for it, I guess it has something to do with the tls version (cloudflare uses 1.2 and google uses 1.3). If you want to see how such a wireshark trace looks like, you can check it out here: https://twitter.com/Sh1bumi/status/1171075999984996355

**Comment by antipatico** [comment]
@shibumi thanks for the effort, but after waiting for a couple of weeks for a response, I ended up using [DNSCrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy), which is much more reliable and consistent.

**Comment by shibumi** [comment]
@antipatico What do you mean with reliable and consistent?
I use Google DNS + systemd-resolved in strict dns over TLS mode, now. And it works super reliable. I have no issues.

I would just prefer using cloudflare, but cloudflare has a bug in their implementation. They use TLS 1.2 instead of TLS 1.3. I have reported it to a cloudflare dev and they are on it.

**Comment by carloabelli** [comment]
@shibumi Do you have any more details on this cloudflare bug? Is this bug simply that systemd-resolved is incompatible with TLS 1.2 and requires TLS 1.3? Seems that systemd-resolved is also not working with AdGuard DNS over TLS either and the wireshark output is suspiciously similar.

**Comment by shibumi** [comment]
@carloabelli Hi, yes I do have more information about this. This bug has been resolved with this PR:

https://github.com/systemd/systemd/pull/13809

You can  find an explanation about this here: https://github.com/systemd/systemd/issues/13528#issuecomment-544274060
